Sources/Datadog/CrashReporting/DDCrashReportingPluginType.swift

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Crash Report format supported by Datadog SDK.
+@objc
+public class DDCrashReport: NSObject {
+    /// The date of the crash occurrence.
+    internal let crashDate: Date?
+    // TODO: RUMM-1053 - consider providing / formatting this characteristic of the crash report
+    internal let signalCode: String?
+    // TODO: RUMM-1053 - consider providing / formatting this characteristic of the crash report
+    internal let signalName: String?
+    // TODO: RUMM-1053 - consider providing / formatting this characteristic of the crash report
+    internal let signalDetails: String?
+    // TODO: RUMM-1053 - consider providing / formatting this characteristic of the crash report
+    internal let stackTrace: String?
+
+    public init(
+        crashDate: Date?,
+        signalCode: String?,
+        signalName: String?,
+        signalDetails: String?,
+        stackTrace: String?
+    ) {
+        self.crashDate = crashDate
+        self.signalCode = signalCode
+        self.signalName = signalName
+        self.signalDetails = signalDetails
+        self.stackTrace = stackTrace
+    }
+}
+
+/// An interface for enabling crash reporting feature in Datadog SDK.
+/// It is implemented by `DDCrashReportingPlugin` from `DatadogCrashReporting` framework.
+@objc
+public protocol DDCrashReportingPluginType: class {
+    /// Reads unprocessed crash report if available.
+    /// - Parameter completion: the completion block called with the value of `DDCrashReport` if a crash report is available
+    /// or with `nil` otherwise. The value returned by the receiver should indicate if the crash report was processed correctly (`true`)
+    /// or something went wrong (`false)`. Depending on the returned value, the crash report will be purged or perserved for future read.
+    func readPendingCrashReport(completion: (DDCrashReport?) -> Bool)
```

Comment:
Are there any expectations (for callers and implementations) on whether it should be synchronous or asynchronous?

Reply:
The note on sync/async + threading assumptions should be definitely there in this public API comment 👍.

Right now it's synchronous, but the `completion` is there because it will be async at the end. The next PR will deliver SDK context injection to crash reports, and that operation will need to be synchronised in `DDCrashReportingPlugin's` implementation (as it might be called from different threads). The same thread will be then used for dispatching `completion: (DDCrashReport?) -> Bool`asynchronously.

I'd prefer to update this comment once a real threading is introduced in next PR. Right now I can only guess how it may look like 🙂. WDYT?

Reply:
That sounds clear and sensible 👍

Reply:
A question that's again probably out of scope and not a big deal now, will there a be some kind of queueing or debouncing if that method is called multiple times before the first call completes?
Especially if the implementation behind it tracks some state beginning with the first call.